engines/wazero/wazero.go

```diff
@@ -263,56 +356,71 @@ func newInvokeContext(ctx context.Context, ic *invokeContext) context.Context {
 	return context.WithValue(ctx, invokeContextKey{}, ic)
 }
 
+// fromInvokeContext returns invokeContext value or nil if there was none.
+//
+// Note: This is never nil if called by Instance.Invoke
+// TODO: It may be better to panic on nil or error as if this is nil, it is a bug in waPC, as no other path calls this.
 func fromInvokeContext(ctx context.Context) *invokeContext {
 	ic, _ := ctx.Value(invokeContextKey{}).(*invokeContext)
-	if ic == nil {
-		ic = &invokeContext{}
```

Comment:
this was confusing. on the read path it doesn't make sense to instantiate this as it will never be re-applied to the context. We should panic if nil as if this is ever nil it is a bug using the instantiated module, and all code that does that should be in this repo, I think, right?